src/ldp/representation/RepresentationPreferences.ts

```diff
@@ -1,12 +1,21 @@
-import type { RepresentationPreference } from './RepresentationPreference';
+/**
+  * Represents preferred values along a single content negotiation dimension.
+  *
+  * The number represents how preferred this value is from 0 to 1.
+  * Follows the quality values rule from RFC 7231:
+  * "The weight is normalized to a real number in the range 0 through 1,
+  * where 0.001 is the least preferred and 1 is the most preferred; a
+  * value of 0 means "not acceptable"."
+  */
+export type ValuePreferences = Record<string, number>;
 
 /**
- * Contains the preferences of which kind of representation is requested.
+ * Contains preferences along multiple content negotiation dimensions.
  */
 export interface RepresentationPreferences {
-  type?: RepresentationPreference[];
-  charset?: RepresentationPreference[];
-  datetime?: RepresentationPreference[];
-  encoding?: RepresentationPreference[];
-  language?: RepresentationPreference[];
+  type?: ValuePreferences;
+  charset?: ValuePreferences;
+  datetime?: ValuePreferences;
+  encoding?: ValuePreferences;
+  language?: ValuePreferences;
```

Comment:
Since the interface is being changed now this might also be a good time to determine what the difference if between an empty ValuePreferences and the key not existing? Perhaps we want the keys to not be optional, or at least document what it means.

Reply:
I thought about that too; at the moment, I left it for convenience of instantiating. Will try changing and see what it gives.

Reply:
Several places use `{}` or `{ type }` to instantiate; would not make sense to change them all. Unless we make `RepresentationPreferences` a class with a constructor.

Reply:
Yea I assumed that might be an issue, but it might make sense to add a comment how to interpret a key not being there vs having an empty object no?

Reply:
Will document indeed!